### PR TITLE
MPN-1558: Create minimal failing test for DataFrameReader.toRDD

### PR DESCRIPTION
**Pull request by yawaramin**
Create a test that demonstrates the error on trying to convert a case
class value into an RDD, as per #2.

### CODE REVIEW

File: core/src/test/scala/com.mediative.sparrow/DataFrameReaderTest.scala [hunk 1]
```diff
@@ -153,12 +168,58 @@ class DataFrameReaderTest extends FreeSpec with BeforeAndAfterAll {
...
+
+        val expected = TestToRdd2(1, Option.empty)
+        val df =
+          sqlContext.createDataFrame(sc.parallelize(List(expected)))
+
+        assert(df.toRDD[TestToRdd2].toOption.get.first == expected)
+      }
+    }
+
+    "round-trip TestToRdd3 from RDD to DataFrame back to RDD" in {
```
Comment (suhailshergill) [inline core/src/test/scala/com.mediative.sparrow/DataFrameReaderTest.scala]: @yawaramin minor comment, can we give better names than `TestToRdd3`. it's not so important when naming the case classes, as in the description of the tests (which should be humanly understandable)

**Comment by suhailshergill** [comment]
merged, because tests succeed locally. 
travis fails, because the it's unable to obtain sbt from the typesafe resolver

**Comment by gpoirier** [comment]
This arguably doesn't show case a `RowConverter` bug. I already explained the issue to Luis. The issue here is that the use case it was trying to solve was reading from JSON/CSV, where there's no date type. So when reading from a date, it's assumed to be a `String`. But when Parquet is used to convert the case class to a `DataFrame`, then dates are saved as a `TimestampType`. The field converter(s) for dates doesn't support reading dates from that type. 

The case classes saved with Parquet either need to use a different `FieldConverter` or the existing `FieldConverter` need to have a different behaviour depending on the field type.

But the additional tests cases here should probably not be defined this way here. The successful tests doesn't add anything beyond what's already tested and this test class already has a structure that differ what's being done in the same spec. If `@schema` was broken, `DataFrameReaderTest` should not be broken too.

**Comment by suhailshergill** [comment]
@gpoirier a couple of points:
* we're not entering the parquet realm in the tests here. they're restricted to `RDD[T] => DataFrame => RDD[T]` realm
* would these be better as integration tests perhaps? or do you have another suggestion?

**Comment by gpoirier** [comment]
- I don't know if it's technically part of Parquet or not, but what is used in this test is the transformation of a case class to a spark `Row`, which is the part that I thought was part of parquet and that is causing all those issues and using Reflection. The `RDD[T]` => `DataFrame` conversion is the layer that is causing issues when using parquet.

- I'm mostly suggesting not to use @schema (it's not related to the issue, the DSL should be used). And it probably would be better fitted to a different spec. But it's not really an integration test, it could even be defined without using Spark at all by creating the `Row` object manually.

**Pull request merged**